### PR TITLE
Make Assignments and Metadata more dict-like

### PR DESCRIPTION
**Pull request by pfhayes**
@alexandraj777

### CODE REVIEW

File: sigopt/objects.py [hunk 1]
```diff
@@ -51,12 +51,28 @@ def __eq__(self, other):
     )
 
 
-class Assignments(ApiObject):
-  def get(self, key):
-    return self._body.get(key)
+class _DictWrapper(ApiObject, dict):
```
Comment (alexandraj777) [inline sigopt/objects.py]: Should we restrict setting attributes on metadata and assignments without copying?

Comment (pfhayes) [inline sigopt/objects.py]: If we make them mutable, users can just edit them in place and then PUT them back if they want to change them. What would the reasoning be for making them immutable?

**Comment by pfhayes** [comment]
Made some more mutability updates

**Comment by alexandraj777** [comment]
👍

**Pull request merged**